components/board/genericlinux/gpio.go

```diff
@@ -220,10 +229,29 @@ func (pin *gpioPin) Close() error {
 }
 
 func gpioInitialize(cancelCtx context.Context, gpioMappings map[int]GPIOBoardMapping,
-	waitGroup *sync.WaitGroup, logger golog.Logger,
-) map[string]*gpioPin {
+	interruptConfigs []board.DigitalInterruptConfig, waitGroup *sync.WaitGroup, logger golog.Logger,
+) (map[string]*gpioPin, map[string]*digitalInterrupt, error) {
+	interrupts := make(map[string]*digitalInterrupt, len(interruptConfigs))
+	for _, config := range interruptConfigs {
+		interrupt, err := createDigitalInterrupt(cancelCtx, config, gpioMappings, waitGroup)
```

Comment:
If something fails and we return an error without creating the board, the server will attempt to create the board again next time the config gets updated. If the interrupts aren't closed at that point, they won't be able to be created anew because they're already open from last time (and are sending data into channels that are not accessible by the new board).